BLUESPAWN-client/src/hunt/RegistryHunt.cpp

```diff
@@ -0,0 +1,180 @@
+#include "util/log/Log.h"
+
+#include <regex>
+#include <unordered_set>
+
+namespace Registry {
+	REG_SZ_CHECK CheckSzEqual = [](std::wstring s1, std::wstring s2){ return s1 == s2; };
+	REG_SZ_CHECK CheckSzNotEqual = [](std::wstring s1, std::wstring s2){ return s1 != s2; };
+	REG_SZ_CHECK CheckSzEmpty = [](std::wstring s1, std::wstring s2){ return s1.length() == 0; };
+	REG_SZ_CHECK CheckSzRegexMatch = [](std::wstring s1, std::wstring s2){ return std::regex_match(s1, std::wregex(s2)); };
+	REG_SZ_CHECK CheckSzRegexNotMatch = [](std::wstring s1, std::wstring s2){ return !std::regex_match(s1, std::wregex(s2)); };
+
+	REG_DWORD_CHECK CheckDwordEqual = [](DWORD d1, DWORD d2){ return d1 == d2; };
+	REG_DWORD_CHECK CheckDwordNotEqual = [](DWORD d1, DWORD d2){ return d1 != d2; };
+
+	REG_BINARY_CHECK CheckBinaryEqual = [](AllocationWrapper s1, AllocationWrapper s2){
+		return s1.CompareMemory(s2);
+	};
+	REG_BINARY_CHECK CheckBinaryNotEqual = [](AllocationWrapper s1, AllocationWrapper s2){
+		return !s1.CompareMemory(s2);
+	};
+	REG_BINARY_CHECK CheckBinaryNull = [](AllocationWrapper s1, AllocationWrapper s2){ return !s1; };
+
+	REG_MULTI_SZ_CHECK CheckMultiSzSubset = [](std::vector<std::wstring> s1, std::vector<std::wstring> s2){
+		std::unordered_set<std::wstring> vals = { s2.begin(), s2.end() };
+		for(auto string : s1){
+			if(vals.find(string) == vals.end()){
+				return false;
+			}
+		}
+		return true;
+	};
+	REG_MULTI_SZ_CHECK CheckMultiSzExclusion = [](std::vector<std::wstring> s1, std::vector<std::wstring> s2){
+		std::unordered_set<std::wstring> vals = { s2.begin(), s2.end() };
+		for(auto string : s1){
+			if(vals.find(string) != vals.end()){
+				return false;
+			}
+		}
+		return true;
+	};
+	REG_MULTI_SZ_CHECK CheckMultiSzEmpty = [](std::vector<std::wstring> s1, std::vector<std::wstring> s2){
+		return s1.size() == 0;
+	};
+
+	RegistryCheck::RegistryCheck(std::wstring wValueName, RegistryType type, std::wstring wData,
+		bool MissingBad, REG_SZ_CHECK check) :
+		value{ wValueName, type, wData },
+		MissingBad{ MissingBad },
+		wCheck{ check }{}
+
+	RegistryCheck::RegistryCheck(std::wstring wValueName, RegistryType type, DWORD dwData,
+		bool MissingBad, REG_DWORD_CHECK check) :
+		value{ wValueName, type, dwData },
+		MissingBad{ MissingBad },
+		dwCheck{ check }{}
+
+	RegistryCheck::RegistryCheck(std::wstring wValueName, RegistryType type, AllocationWrapper lpData,
+		bool MissingBad, REG_BINARY_CHECK check) :
+		value{ wValueName, type, lpData },
+		MissingBad{ MissingBad },
+		lpCheck{ check }{}
+
+	RegistryCheck::RegistryCheck(std::wstring wValueName, RegistryType type, std::vector<std::wstring> vData,
+		bool MissingBad, REG_MULTI_SZ_CHECK check) :
+		value{ wValueName, type, vData },
+		MissingBad{ MissingBad },
+		vCheck{ check }{}
+
+	RegistryType RegistryCheck::GetType() const {
+		return value.type;
+	}
+
+	std::vector<RegistryValue> CheckValues(const RegistryKey& key, const std::vector<RegistryCheck> checks){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'checks' should be passed by const reference.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=4967854)